docs/guide/start-identity-permissions-and-scopes.md

```diff
@@ -0,0 +1,201 @@
+
+
+Yii2-Oauth2-Server Integration
+------------------------------
+
+The Yii2-Oauth2-Server integrates with the Yii2 application by taking care of the entire Oauth 2.0 flow.
+After successful authorization, when the Client makes server requests with a valid authorization header
+the Yii2-Oauth2-Server will set the Yii2 "user identity class" (`Yii::$app->user->identity`) for the "user component"
+(`Yii::$app->user`). 
+
+### Identity and RBAC usage
+
+Let's say that in our example there is a backend application (from now on abbreviated as BEA) has 2 roles:
+'user' and 'admin'.
+When a User authorizes a Client, let's say a frontend application(from now on abbreviated as BEA), to use the BEA,
+all calls that are made by that Client will be automatically (under the condition that the Client correctly sends the 
+"authentication" header) done with that user's identity.
+
+Let's say that whe have two users: Alice is "admin" and user Bob is a regular "user".
+
+When Alice authorizes the FEA via the Authorization Code Grand the FEA can use the "bearer token" authorization header
+to make API calls against the BEA. For all these calls the `Yii::$app->user->identity` will be set to the user identity
+of Alice. When Bob would authorize the FEA all calls with his bearer token by the FEA will set the
+`Yii::$app->user->identity` to Bob's identity.
+
+Since the identity is set, you can simply use `Yii::$app->user->can('role_you_want_to_check')` anywhere you want.
+E.g. `Yii::$app->user->can('admin')` would return `true` for Alice and `false` for Bob.
+
+> Note: Until now "scope" is never used and probably is not needed. 
+  Only when we have multiple clients, let's say FEA1 and FEA2, that need a different access scope for the *same* user,
+  scopes come into play.
+
+### Scope usage
+
+> Scopes only come into play in delegation scenarios, and always *limit* what a Client can do on *behalf* of a user:
+  a scope cannot allow an application to do more than what the user can do.
+
+Let's say (continuing from our sample above) that Bob (with the "user" role) wants to authorize two different clients:
+- Your own front-end application (FEA) to read, create and delete emails on his behalf.
+- A third-party app (from now on abbreviated as 3rdPA) to only create emails on his behalf.
+
+In that case you would need to define the scopes 'read_email', 'create_email' and 'delete_email' and assign them to the
+different Clients. Then Bob can specify which scopes the Client may use on his behalf during the authorization of the 
+Client.
+
+Note that another user, let's say Charles (who also has the role "user"), can allow different scopes for the same
+clients, e.g.:
+- FEA1 may read and create emails on his behalf.
+- 3rdPA may create and also delete emails on his behalf.
+
+When scopes are used you can use `Yii::$app->getModule('oauth2')->requestHasScope('scope_you_want_to_check')` to see if
+the client that makes the request has the required scope.
+
```

Comment:
Think of it as role limiting was done above, but now it applies to client app not the user account!

Reply:
Thanks, nice addition 👍